src/dbnode/client/host_queue.go

```diff
@@ -43,24 +44,29 @@ type queue struct {
 	sync.WaitGroup
 	sync.RWMutex
 
-	opts                                       Options
-	nowFn                                      clock.NowFn
-	host                                       topology.Host
-	connPool                                   connectionPool
-	writeBatchRawRequestPool                   writeBatchRawRequestPool
-	writeBatchRawRequestElementArrayPool       writeBatchRawRequestElementArrayPool
-	writeTaggedBatchRawRequestPool             writeTaggedBatchRawRequestPool
-	writeTaggedBatchRawRequestElementArrayPool writeTaggedBatchRawRequestElementArrayPool
-	workerPool                                 xsync.PooledWorkerPool
-	size                                       int
-	ops                                        []op
-	opsSumSize                                 int
-	opsLastRotatedAt                           time.Time
-	opsArrayPool                               *opArrayPool
-	drainIn                                    chan []op
-	writeOpBatchSize                           tally.Histogram
-	fetchOpBatchSize                           tally.Histogram
-	status                                     status
+	opts                                         Options
+	nowFn                                        clock.NowFn
+	host                                         topology.Host
+	connPool                                     connectionPool
+	writeBatchRawRequestPool                     writeBatchRawRequestPool
+	writeBatchRawV2RequestPool                   writeBatchRawV2RequestPool
+	writeBatchRawRequestElementArrayPool         writeBatchRawRequestElementArrayPool
+	writeBatchRawV2RequestElementArrayPool       writeBatchRawV2RequestElementArrayPool
+	writeTaggedBatchRawRequestPool               writeTaggedBatchRawRequestPool
+	writeTaggedBatchRawV2RequestPool             writeTaggedBatchRawV2RequestPool
+	writeTaggedBatchRawRequestElementArrayPool   writeTaggedBatchRawRequestElementArrayPool
+	writeTaggedBatchRawV2RequestElementArrayPool writeTaggedBatchRawV2RequestElementArrayPool
+	workerPool                                   xsync.PooledWorkerPool
+	size                                         int
+	ops                                          []op
+	opsSumSize                                   int
+	opsLastRotatedAt                             time.Time
+	opsArrayPool                                 *opArrayPool
+	drainIn                                      chan []op
+	writeOpBatchSize                             tally.Histogram
+	fetchOpBatchSize                             tally.Histogram
+	status                                       status
+	serverSupportsV2APIs                         bool
```

Comment:
Similar comment to above regarding versioning. Imagining a `serverSupportsV3APIs` and `serverSupportsV4APIs` tag here later on is quite painful.

Reply:
I think for this portion I'll leave as is as a bool because with an enum you need to handle the case where its an invalid value. Hopefully we can just delete this code soon